tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPersonNameFormatter.java

```diff
@@ -101,6 +100,38 @@ public void TestBasic() {
         checkFormatterData(personNameFormatter);
     }
 
+public void TestLiteralCollapsing1() {
+    ImmutableMap<ULocale, Order> localeToOrder = ImmutableMap.of(); // don't worry about using the order from the locale right now.
+
+    NamePatternData namePatternData2 = new NamePatternData(
+        localeToOrder,
+        "order=givenFirst",     "⓵{prefix}⓶{given}⓷{given2}⓸{surname}⓹{surname2}⓺{suffix}⓻",
+        "order=surnameFirst",   "⓵{surname-allCaps}⓶{surname2}⓷{prefix}⓸{given}⓹{given2}⓺{suffix}⓻",
+        "order=sorting",        "⓵{surname}⓶{surname2},⓷{prefix}⓸{given}⓹{given2}⓺{suffix}⓻");
+
+    PersonNameFormatter personNameFormatter2 = new PersonNameFormatter(namePatternData2, FALLBACK_FORMATTER);
+
+    check(personNameFormatter2, sampleNameObject5, "order=givenFirst", "Mary⓷⓸Smith"); // reasonable, given no break
+    check(personNameFormatter2, sampleNameObject5, "order=surnameFirst", "⓵SMITH⓶⓸Mary"); // reasonable, given no break
+    check(personNameFormatter2, sampleNameObject5, "order=sorting", "⓵Smith⓶⓸Mary"); // should be something like Smith,⓷❸Mary
+}
+
+public void TestLiteralCollapsing2() {
+    ImmutableMap<ULocale, Order> localeToOrder = ImmutableMap.of(); // don't worry about using the order from the locale right now.
+
+    NamePatternData namePatternData2 = new NamePatternData(
+        localeToOrder,
+        "order=givenFirst",     "❶{prefix}⓶ ❷{given}⓷ ❸{given2}⓸ ❹{surname}⓹ ❺{surname2}⓺ ❻{suffix}❼",
+        "order=surnameFirst",   "❶{surname-allCaps}⓶ ❷{surname2}⓷ ❸{prefix}⓸ ❹{given}⓹ ❺{given2}⓺ ❻{suffix}❼",
+        "order=sorting",        "❶{surname}⓶ ❷{surname2}, ❸{prefix}⓸ ❹{given}⓹ ❺{given2}⓺ ❻{suffix}❼");
+
+    PersonNameFormatter personNameFormatter2 = new PersonNameFormatter(namePatternData2, FALLBACK_FORMATTER);
+
+    check(personNameFormatter2, sampleNameObject5, "order=givenFirst", "Mary⓷ ❹Smith"); // reasonable
+    check(personNameFormatter2, sampleNameObject5, "order=surnameFirst", "❶SMITH⓶ ❹Mary"); // reasonable
+    check(personNameFormatter2, sampleNameObject5, "order=sorting", "❶Smith⓶ ❹Mary"); // should be something like Smith,❸Mary
+}
+
```

Comment:
These two new tests (TestLiteralCollapsing1() and TestLiteralCollapsing2() ) duplicate my own test (TestLiteralTextElision() ) below.  I think you're doing some things I'm not, but I think it'd be good to agree on a name, get rid of any redundant test code, and put all of the tests closer together in the source file.